pkg/compactor/block_upload.go

```diff
@@ -518,6 +528,10 @@ func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.U
 		}
 	}
 
+	if err := c.validateMaximumBlockSize(logger, blockMetadata.Thanos.Files, userID); err != nil {
```

Comment:
Why do we do this validation again? The only way how block metadata is uploaded is via start handler, and we already check block size there.

Also it's little weird to do this right after downloading the block to local disk or using `SizeBytes` field to check file sizes.

If we end up with only doing block size validation in `sanitizeMeta`, I'd suggest to get rid of `validateMaximumBlockSize` method, and merge it into sanitizeMeta instead (we already iterate over files there, check for negative file `SizeBytes`, so the only missing piece is to check total block size against `CompactorBlockUploadMaxBlockSizeBytes`)

Reply:
> Why do we do this validation again? The only way how block metadata is uploaded is via start handler, and we already check block size there.

Imagine an upload is started before the limit is applied, but a block is too big for the compactor to handle. Setting this limit is a natural response to that, but if we didn't limit on validation as well then it wouldn't help us.

> Also it's little weird to do this right after downloading the block to local disk or using SizeBytes field to check file sizes.

Good catch! This will require more reordering to get the block metadata in earlier then.

> If we end up with only doing block size validation in sanitizeMeta

I don't see this being the case due to the above reasoning about doing this validation both when starting + validating the upload.

Reply:
> Imagine an upload is started before the limit is applied, but a block is too big for the compactor to handle. Setting this limit is a natural response to that, but if we didn't limit on validation as well then it wouldn't help us.

That's a good point.